src/com/dotmarketing/business/cache/provider/caffine/CaffineCache.java

```diff
@@ -0,0 +1,287 @@
+            keys = m.keySet();
+        }
+
+        return keys;
+    }
+
+    @Override
+    public List<Map<String, Object>> getStats() {
+
+        List<Map<String, Object>> list = new ArrayList<>();
+
+        Set<String> currentGroups = new HashSet<>();
+        currentGroups.addAll(getGroups());
+
+        Cache<String, Object> defaultCache = getCache(DEFAULT_CACHE);
+        for (String group : currentGroups) {
+
+            Map<String, Object> stats = new HashMap<>();
+            stats.put("name", getName());
+            stats.put("key", getKey());
+            stats.put("cache", getCache(group));
+            stats.put("region", group);
+            stats.put("toDisk", false);
+
+            Cache<String, Object> foundCache = getCache(group);
+            stats.put("memory", foundCache.estimatedSize());
+            stats.put("CacheStats", foundCache.stats());
+            stats.put("disk", -1);
+
+            boolean isDefault = (!DEFAULT_CACHE.equals(group) && foundCache.equals(defaultCache));
+            stats.put("isDefault", isDefault);
+
+            int configured = isDefault ? Config.getIntProperty("cache." + DEFAULT_CACHE + ".size")
+                    : (Config.getIntProperty("cache." + group + ".size", -1) != -1)
+                            ? Config.getIntProperty("cache." + group + ".size")
+                            : (group.startsWith(WORKING_CACHE_PREFIX) && Config.getIntProperty(
+                                    "cache." + WORKING_CACHE_PREFIX + ".size", -1) != -1)
+                                            ? Config.getIntProperty(
+                                                    "cache." + WORKING_CACHE_PREFIX + ".size")
+                                            : (group.startsWith(LIVE_CACHE_PREFIX)
+                                                    && Config.getIntProperty(
+                                                            "cache." + LIVE_CACHE_PREFIX + ".size",
+                                                            -1) != -1)
+                                                                    ? Config.getIntProperty(
+                                                                            "cache." + LIVE_CACHE_PREFIX
+                                                                                    + ".size")
+                                                                    : Config.getIntProperty(
+                                                                            "cache." + DEFAULT_CACHE
+                                                                                    + ".size");
+            stats.put("configuredSize", configured);
+
+            list.add(stats);
+        }
+
+        return list;
+    }
+
+    @Override
+    public void shutdown() {
+        Logger.info(this.getClass(), "===== Calling shutdown [" + getName() + "].");
+        isInitialized = false;
+    }
+
+    private Cache<String, Object> getCache(String cacheName) {
```

Comment:
We got together and determined that refactoring to abstract the repeated code does not bring benefit. Leave as it is.